lib/adapters/http.js

```diff
@@ -664,7 +664,8 @@ function HttpPouch(opts, callback) {
 
       var keysAsString =
         'keys=' + encodeURIComponent(JSON.stringify(opts.keys));
-      if (keysAsString.length + params.length + 1 <= MAX_URL_LENGTH) {
+      // FIX for sync gateway, which does not accept multiple keys on a GET request
+      if (false) { //keysAsString.length + params.length + 1 <= MAX_URL_LENGTH) {
```

Comment:
I think I experimented and thus found out. SyncGateway only implements the replication protocol, not the full CouchDB API.
Try querying your SG instance: `http://...:5984/db/_all_docs?keys=["id1","id2"]` and see if it is implemented yet.